### ISSUE TITLE
Postgres: If you have subpartitions of a table, it generates types for each subpartition table, not the parent table.

### ISSUE THREAD

**Issue by saddiqs1** [RobinBlomberg/kysely-codegen#76]
Imagine the following schema:

**`info`**
| column | type |
| ---------- | ---- |
| id | SERIAL PRIMARY KEY|
| name | TEXT    |

I have multiple partitions of this table, e.g. `info_1`, `info_2` ...

When I generate the types, I expect to get the types of `info` since I query directly on `info`, however I get types generated for `info_1` and `info_2`. In this example this is not an issue, however in my application I have over 200+ partitions, and I want to query on the parent table for it to allow postgres to handle which partition it targets.

## Upvote & Fund

- We're using [Polar.sh](https://polar.sh/RobinBlomberg) so you can upvote and help fund this issue.
- We receive the funding once the issue is completed & confirmed by you.
- Thank you in advance for helping prioritize & fund our backlog.

<a href="https://polar.sh/RobinBlomberg/kysely-codegen/issues/76">
<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://polar.sh/api/github/RobinBlomberg/kysely-codegen/issues/76/pledge.svg?darkmode=1">
  <img alt="Fund with Polar" src="https://polar.sh/api/github/RobinBlomberg/kysely-codegen/issues/76/pledge.svg">
</picture>
</a>

**Comment by RobinBlomberg** [comment]
Hey! I would like to thank whoever funded this issue, and I will make sure to look at this as soon as possible. I have not really worked with subpartitions in Postgres, so I will have to do some experimentation, and also determine if this is a kysely or kysely-codegen issue. Stay tuned!

**Comment by RobinBlomberg** [comment]
Should be released in [kysely-codegen@0.16.0](https://github.com/RobinBlomberg/kysely-codegen/releases/tag/0.16.0)! 🚀

**Comment by sglim** [comment]
Hi @RobinBlomberg. I found that kysely-codegen@0.16.0 excludes all the partitions, but it still doesn't create the partitioned table itself. Do you have any solution for generating the missing partition table?

**Comment by apuchitnis** [comment]
Hey @saddiqs1! Did upgrading the library address the issue for you? Or did you run into the same issue as [above](https://github.com/RobinBlomberg/kysely-codegen/issues/76#issuecomment-2521803345).

**Comment by saddiqs1** [comment]
Hey, been a while since I've looked at this! Just had a quick test with the most up to date version (`0.18.5`) and it seems to do what @sglim mentions, by default all partitions are excluded. When you pass in the `--partitions` option when running the command, the original issue persists. 

So as far as I can tell, you can either exclude partitions completely, or include every partition in your DB. The ideal solution would've been a middle ground between these 2 - being able to get the types for just the parent table alone.

**Comment by RobinBlomberg** [comment]
Do you have a way to reproduce this? The `measurement` table below is the one you want to be generated, right? It's possible that I'm missing something from the description, as it seems to be working as expected for me.

```sql
CREATE TABLE measurement (
    city_id         int not null,
    logdate         date not null,
    peaktemp        int,
    unitsales       int
) PARTITION BY RANGE (logdate);

CREATE TABLE measurement_y2006m02 PARTITION OF measurement
    FOR VALUES FROM ('2006-02-01') TO ('2006-03-01');

CREATE TABLE measurement_y2006m03 PARTITION OF measurement
    FOR VALUES FROM ('2006-03-01') TO ('2006-04-01');
```

| Without `--partitions` | With `--partitions` |
| - | - |
| <img width="558" alt="Image" src="https://github.com/user-attachments/assets/97b56332-8ffa-44b2-80f9-368f291bb3ea" /> | <img width="558" alt="Image" src="https://github.com/user-attachments/assets/cc4bc10e-4f74-4c6e-9a4c-5db99f825444" /> |

**Comment by mariandaniel24** [comment]
Encountered an issue where `--partitions` flag wouldn't generate the parent table, turns out the problem was that were running an  older version of `kysely`.
Fixed it by installing `kysely@0.28.5` with `kysely-codegen@0.19.0`.
